components/identity/utils/useIdentity.ts

```diff
@@ -0,0 +1,114 @@
+import { get, update } from 'idb-keyval'
+import { hexToString, isHex } from '@polkadot/util'
+import { GenericAccountId } from '@polkadot/types/generic/AccountId'
+import { onApiConnect } from '@kodadot1/sub-api'
+
+import { emptyObject } from '@/utils/empty'
+import { identityStore } from '@/utils/idbStore'
+import shortAddress from '@/utils/shortAddress'
+
+type Address = string | GenericAccountId | undefined
+type IdentityFields = Record<string, string>
+
+const resolveAddress = (account: Address): string => {
+  return account instanceof GenericAccountId
+    ? account.toString()
+    : account || ''
+}
+
+const handleRaw = (display): string => {
+  if (display?.isRaw) {
+    return display.asRaw.toHuman() as string
+  }
+
+  if (isHex(display?.Raw)) {
+    return hexToString(display?.Raw)
+  }
+
+  return display?.toString()
+}
+
+const fetchIdentity = async (address: string): Promise<IdentityFields> => {
+  if (!address) {
+    return emptyObject<IdentityFields>()
+  }
+
+  const { apiInstance } = useApi()
+  const api = await apiInstance.value
+  const optionIdentity = await api?.query.identity?.identityOf(address)
+  const identityFresh = optionIdentity?.unwrapOrDefault()
+
+  if (!identityFresh?.size) {
+    return emptyObject<IdentityFields>()
+  }
+
+  const final = Array.from(identityFresh.info)
+    .filter(([, value]) => !Array.isArray(value) && !value.isEmpty)
+    .reduce((acc, [key, value]) => {
+      acc[key] = handleRaw(value)
+      return acc
+    }, {} as IdentityFields)
```

Comment:
Not the first time I see this array filter in codebase

```suggestion
    }, {}) as IdentityFields
```
better?